src/Common/src/System/Net/HttpDateParser.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Globalization;
+
+namespace System.Net
+{
+    internal static class HttpDateParser
+    {
+        private static readonly string[] s_dateFormats = new string[] {
+            "r", // RFC 1123, required output format but too strict for input
+            "ddd, d MMM yyyy H:m:s 'GMT'", // RFC 1123 (r, except it allows both 1 and 01 for date and time)
+            "ddd, d MMM yyyy H:m:s 'UTC'", // RFC 1123, UTC
+            "ddd, d MMM yyyy H:m:s", // RFC 1123, no zone - assume GMT
+            "d MMM yyyy H:m:s 'GMT'", // RFC 1123, no day-of-week
+            "d MMM yyyy H:m:s 'UTC'", // RFC 1123, UTC, no day-of-week
+            "d MMM yyyy H:m:s", // RFC 1123, no day-of-week, no zone
+            "ddd, d MMM yy H:m:s 'GMT'", // RFC 1123, short year
+            "ddd, d MMM yy H:m:s 'UTC'", // RFC 1123, UTC, short year
+            "ddd, d MMM yy H:m:s", // RFC 1123, short year, no zone
+            "d MMM yy H:m:s 'GMT'", // RFC 1123, no day-of-week, short year
+            "d MMM yy H:m:s 'UTC'", // RFC 1123, UTC, no day-of-week, short year
+            "d MMM yy H:m:s", // RFC 1123, no day-of-week, short year, no zone
+
+            "dddd, d'-'MMM'-'yy H:m:s 'GMT'", // RFC 850
+            "dddd, d'-'MMM'-'yy H:m:s 'UTC'", // RFC 850, UTC
+            "dddd, d'-'MMM'-'yy H:m:s zzz", // RFC 850, offset
+            "dddd, d'-'MMM'-'yy H:m:s", // RFC 850 no zone
+            "ddd MMM d H:m:s yyyy", // ANSI C's asctime() format
+
+            "ddd, d MMM yyyy H:m:s zzz", // RFC 5322
+            "ddd, d MMM yyyy H:m:s", // RFC 5322 no zone
+            "d MMM yyyy H:m:s zzz", // RFC 5322 no day-of-week
+            "d MMM yyyy H:m:s", // RFC 5322 no day-of-week, no zone
```

Comment:
This list of format is now larger than System.Net.Http was using in the past.  Why?  Isn't that a breaking change?

Reply:
It's bigger than System.Net.Http, but that was to bring it up to parity with System.Net.Request.  Most of the new ones are enabling using `'UTC'` instead of `'GMT'` (and one - the explicit `'r'` - could actually be removed).

If I was going to consolidate the date parsing, I was going to have to break _something_, and I figured that widening the valid parse cases which previously couldn't be passed would be less harmful than throwing errors where none occurred before.  I suppose I could have separate sets for each.... is that really the way we want to go?

Reply:
> is that really the way we want to go?

@davidsh and @karelz should comment on that.  But as it stands, this PR is more than just about consolidating code; it's changing behavior.

Reply:
Yes, some aspects of this are a breaking change. But it moves the parser toward something better and embraces some newer RFCs. Generally, it allows for more flexible parsing of dates. 

In some cases, it fixes some bugs where dates are parsed wrong.

>Drops ability to use non-matching date/day-of-week strings (ie "Sat, 25 Mar 2018 16:33:01 GMT" - the 25th was actually a Sunday)

Overall, I think this will be a good change.

Reply:
Ok